src/gt4py/stencil_builder.py

```diff
@@ -18,6 +18,7 @@
 from typing import TYPE_CHECKING, Any, Dict, Optional, Type, Union
 
 import gt4py.caching
+import gt4py.frontend
```

Comment:
Good catch! Out of the scope of this PR, but I wonder if it is actually needed since the error hasn't been found earlier, and if it is indeed needed, why is not tested...

Reply:
Yes I think this is needed. Check the error message above for the circular import error.